### PR TITLE
feat: setup truffle smart contracts folder

### PR DESCRIPTION
**Pull request by abretonc7s**
Setup and integrate basic projects to integrate smart contracts validation with sdk.

### CODE REVIEW

File: packages/devnext/src/pages/index.tsx [hunk 1]
```diff
@@ -146,6 +149,55 @@ export default function Home() {
...
+    }
+
+    const provider = new ethers.providers.JsonRpcProvider(rpcUrl);
+    const contract = new ethers.Contract(contractAddress, SimpleABI.abi, provider);
+
+    const text = await contract.ping();
+    console.debug('ping', text)
+
+    const chainId = await provider.getNetwork()
+    console.debug(`chainId`, chainId)
```
Comment (omridan159) [inline packages/devnext/src/pages/index.tsx:171]:
```suggestion
try {
  const text = await contract.ping();
  console.debug('ping', text);
  
  const network = await provider.getNetwork();
  console.debug('Network', network);
} catch (error) {
  console.error('Error pinging ethers:', error.message);
}
```

File: packages/devnext/src/pages/index.tsx [hunk 2]
```diff
@@ -146,6 +149,55 @@ export default function Home() {
...
+    console.log(`chainId`, chainId)
+
+    const contract = getContract({
+      address: contractAddress,
+      abi: SimpleABI.abi,
+      publicClient: client,
+    })
+
+    const text = await contract.read.ping()
+    console.debug(`ping`, text)
```
Comment (omridan159) [inline packages/devnext/src/pages/index.tsx:198]:
```suggestion
try {
  const balance = await client.getBalance({ address: BALANCE_ADDRESS });
  console.debug('balance', balance);
  
  const chainId = await client.getChainId();
  console.debug('chainId', chainId);
  
  const contract = getContract({
    address: contractAddress,
    abi: SimpleABI.abi,
    publicClient: client,
  });
  
  const text = await contract.read.ping();
  console.debug('ping', text);
} catch (error) {
  console.error('Error pinging Viem:', error.message);
}
```

File: packages/truffle/src/index.ts
Comment (omridan159) [inline packages/truffle/src/index.ts:1]: Empty file?

Comment (abretonc7s) [reply packages/truffle/src/index.ts:1]: yes, unused for now. Might be later but prevent typescript error if there is no files

**Review by omridan159 — APPROVED** [review]
LGTM

**Pull request merged**